components/src/core/drawer/drawer-layout/drawer-layout.component.ts

```diff
@@ -0,0 +1,38 @@
+import { ChangeDetectionStrategy, Component, Input, ViewEncapsulation, Output, EventEmitter } from '@angular/core';
+
+type VariantType = 'permanent' | 'persistent' | 'temporary';
+
+@Component({
+    selector: 'pxb-drawer-layout',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    template: `
+        <div class="pxb-drawer-layout">
+            <mat-sidenav-container class="pxb-side-nav-container" (backdropClick)="closeDrawer()">
+            <mat-sidenav [mode]="variant === 'temporary' ? 'over' : 'side'" class="px-blue-side-nav" [ngClass]="
+                [variant === 'permanent'
```

Comment:
Use this syntax: `[class.{className}]={conditional}`, example:

<mat-sidenav [class.pxb-drawer-permanent]="variant==='permanent'" ... >

Reply:
The current class rules are very difficult for me to understand.